test/modern_bpf/test_suites/syscall_enter_suite/mmap_e.cpp

```diff
@@ -0,0 +1,63 @@
+#include "../../event_class/event_class.h"
+
+#ifdef __NR_mmap
+
+#include <sys/mman.h>
+
+TEST(SyscallEnter, mmapE)
+{
+	auto evt_test = get_syscall_event_test(__NR_mmap, ENTER_EVENT);
+
+	evt_test->enable_capture();
+
+	/*=============================== TRIGGER SYSCALL  ===========================*/
+
+	void *mock_addr = NULL;
+	size_t mock_length = 1024;
+	int mock_prot = PROT_EXEC | PROT_READ;
+	int mock_flags = MAP_SHARED;
+	int mock_fd = 3;
+	off_t mock_offset = 1023;
+
+	assert_syscall_state(SYSCALL_FAILURE, "mmap", syscall(__NR_mmap, mock_addr, mock_length, mock_prot, mock_flags, mock_fd, mock_offset));
+
+	/*=============================== TRIGGER SYSCALL  ===========================*/
+
+	evt_test->disable_capture();
+
+	evt_test->assert_event_presence();
+
+	if(HasFatalFailure())
+	{
+		return;
+	}
+
+	evt_test->parse_event();
+
+	evt_test->assert_header();
+
+	/*=============================== ASSERT PARAMETERS  ===========================*/
+
+	/* Parameter 1: addr (type: PT_UINT64) */
+	evt_test->assert_only_param_len(1, sizeof(uint64_t));
```

Comment:
Impossible to know the value of this parameter deterministically (it is an address).

Reply:
here we can check against the address we used in the syscall